### PR TITLE
Gh-305: Document Export to CSV

### LINKED ISSUE #305: Document new csv export features
Document the [ToCsv](https://github.com/gchq/Gaffer/issues/2675) Operation with CsvFormats as well as the [ExportToLocalFile](https://github.com/gchq/Gaffer/issues/2707) Operation. An example OperationChain should also be shown that uses both to export elements to a csv file.

### PR DESCRIPTION
**Pull request by p3430233**
Example showing the export of elements from Gaffer's Road Traffic Example into a local openCypher formatted CSV file. The file format used in the example is compatible with neo4j.

# Related Issue

- Resolve #305

### CODE REVIEW

File: docs/ExportToCsv.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+This operation can be chained together with ExportToLocalFile, which will take the iterable of strings and write them to a 
+local file. The ToCsv operation requires a CsvGenerator, this can be built by the user, alternatively it can be autogenerated by 
+supplying the operation with a CsvFormat. CsvFormats that are currently supported include those used by Neo4j and AWS's 
+Neptune graphs, both of which are based on the openCypher [format](./openCypher) but have slight differences.
+
+
+## Example
+The following example shows how to build a chained operation in order to export data from a Gaffer graph and write it to a local file.
+In this instance the example will be outputting the elements from Gaffer's Road Traffic Example into an openCypher formatted CSV file
+compatible with neo4j.
```
Comment (GCHQDeveloper314) [inline docs/ExportToCsv.md]:
```suggestion
compatible with Neo4j.
```

File: docs/ExportToCsv.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Export To Csv
+
+The ToCsv operation allows a user to export data from a running Gaffer instance to an iterable of CSV formatted strings.
+This operation can be chained together with ExportToLocalFile, which will take the iterable of strings and write them to a 
+local file. The ToCsv operation requires a CsvGenerator, this can be built by the user, alternatively it can be autogenerated by 
+supplying the operation with a CsvFormat. CsvFormats that are currently supported include those used by Neo4j and AWS's 
+Neptune graphs, both of which are based on the openCypher [format](./openCypher) but have slight differences.
```
Comment (GCHQDeveloper314) [inline docs/ExportToCsv.md]: There doesn't seem to be other documentation for what CsvFormats are. Maybe there could be some more mention of them or the classes used?

**Review by GCHQDeveloper314 — COMMENTED** [review]
The example is good but some more detail on the CSV format would be useful.

File: docs/ExportToCsv.md (hunk 1 above)
Comment (GCHQDeveloper314) [reply docs/ExportToCsv.md]: Marked as resolved, but I don't see the solution?

File: docs/ExportToCsv.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+## Example
+The following example shows how to build a chained operation in order to export data from a Gaffer graph and write it to a local file.
+In this instance the example will be outputting the elements from Gaffer's Road Traffic Example into an openCypher formatted CSV file
+compatible with neo4j.
+
+
+```
```
Comment (t92549) [inline docs/ExportToCsv.md:16]: This json needs indenting properly

File: docs/ExportToCsv.md (hunk 2 above)
Comment (t92549) [reply docs/ExportToCsv.md]: Could even make sub pages for each format and describe what each header looks like, post an example file, and link to the appropriate reference page (neo4j or Neptune)

**Comment by GCHQDeveloper314** [comment]
Don't worry about the page name or locations. I'm in the process or refactoring the docs structure so they will be changed later. It's the content which counts.

**Review by GCHQDeveloper314 — CHANGES_REQUESTED** [review]
Some files from the assets folder seem to have been removed by mistake.

**Review by GCHQDeveloper314 — APPROVED** [review]
The formatting is not ideal but I think the content is ok. Formatting will need to be reworked later anyway to fit with the general restructuring, so no need for further changes in this PR.

**Pull request merged**